### DISCUSSION TITLE
HTTP request and response

### DISCUSSION THREAD

**Discussion by evercf** [cesanta/mongoose#2187] [Ask questions, get answers] [answered]
Have you ever tried using mg_http_parse and mg_http_reply separately in the main function, because I only want to parse the http request and then send the response data after parsing it
I see mg_http_reply these functions have parameter struct mg_connection *c, I wonder if there will be any problem to use it separately

**Comment by scaprile** [comment, marked as answer]
A function is a function is a function
There are no hidden states
There is no concept of "separate" or "together"
